### PR TITLE
Fix a bunch of issues with logging

### PR DESCRIPTION
**Pull request by sjmonson**
## Summary

Console logging was completely broken when used with the Live progress. This PR fixes a bunch of issues that prevented it from working. Additionally logging due to external packages in workers should also be handled properly (e.g. vLLM)

## Details

There were a bunch of reasons logging was not working correctly:

### 1. `loguru` grabbed an instance of stderr before Rich could redirect it

`loguru`, package we use for logging, is initialized early into the package load and by default will grab a copy of `sys.stderr`. Later when Rich would load, it would replace `sys.stderr` (and `sys.stdout`) with a custom class called `FileProxy` which would redirect writes though itself. Since `loguru` loaded earlier it did not have this redirect in place.

### 2. Spawned worker processes did not receive Rich's `FileProxy`

 TODO Write explaination

### 3. `loguru` not properly configured for multiprocess logging

When initializing non-fork workers `loguru` requires that its state be passed as an argument to the worker in order to share logging queues. This does not matter as much for console logs since the logger can just be initialized again with the stderr pipe in (2). However for log files this causes each worker to open the same file separately. 

## Test Plan

The best way to test this PR is with the following patch which will add a bunch of extra logging from both the main process and workers:

```diff
diff --git a/src/guidellm/scheduler/worker.py b/src/guidellm/scheduler/worker.py
index ece7f587..1819ba81 100644
--- a/src/guidellm/scheduler/worker.py
+++ b/src/guidellm/scheduler/worker.py
@@ -327,6 +327,7 @@ class WorkerProcess(Generic[RequestT, ResponseT]):
 
             # Main loop; loop until canceled
             while True:
+                logger.info("Getting next request to process for worker {}", self.worker_index)
                 await async_semaphore.acquire()
                 request_time = await self.strategy.next_request_time(
                     worker_index=self.worker_index
diff --git a/src/guidellm/scheduler/worker_group.py b/src/guidellm/scheduler/worker_group.py
index 2a3caa2e..21f879f5 100644
--- a/src/guidellm/scheduler/worker_group.py
+++ b/src/guidellm/scheduler/worker_group.py
@@ -746,6 +746,7 @@ class WorkerGroupState(Generic[RequestT, ResponseT]):
     def _update_state_request_counts(self, info: RequestInfo):
         finalized = time.time()
 
+        logger.info("Received request update: {} (status={})", info.request_id, info.status)
         if info.status == "queued":
             self._queued_request_ids.add(info.request_id)
             self._state.queued_requests = len(self._queued_request_ids)
```

Additionally test edge cases such as redirecting console output though a pipe and vLLM backends.

## Related Issues

- Resolves #

---

- [x] "I certify that all code in this PR is my own, except as noted below."

## Use of AI

- [x] Includes code generated or substantially modified by an AI agent
- [x] Includes tests generated or substantially modified by an AI agent

> NOTE: the `Generated-by` or `Assisted-by` trailers should be used in git commit messages when code or tests were generated or substantially modified by an AI agent, as described in the project's [`DEVELOPING.md`](https://github.com/vllm-project/guidellm/blob/main/DEVELOPING.md) file.

---

# git log

commit 5305ccd6dec8665710b2fa98e4f4739f4df1a177
Author: Samuel Monson <smonson@redhat.com>
Date:   Mon Aug 31 17:22:34 2026 -0400

    Redirect worker stdout/stderr though pipe
    
    Pump worker stdout and stderr through a pipe to the main process in
    order to capture it with Rich's fileproxy. This prevents logs emitted by
    external packages from disrupting the Live console output. There are
    still some issues with the loguru logger after this change.
    
    Assisted-by: Claude Code Opus 4.6
    Signed-off-by: Samuel Monson <smonson@redhat.com>

commit 0219e375db1401686532d01f60651718682a908b
Author: Samuel Monson <smonson@redhat.com>
Date:   Mon Aug 31 17:38:10 2026 -0400

    Rework logger to fix Rich interaction and MP
    
    Fixes 3 main issues:
    
    1. Loguru no longer bypasses Rich's Live console redirection
    2. Spawned child processes now inherit the parent's logging instance
    3. Theaded logging is has been enabled for improved performance
    
    Assisted-by: Cursor
    Signed-off-by: Samuel Monson <smonson@redhat.com>

commit 170f7939466e6b9fbeebcf7135cd20d3249a2674
Author: Samuel Monson <smonson@redhat.com>
Date:   Wed Sep 2 15:47:46 2026 -0400

    Only redirect stderr if it would interfere with stdout
    
    If a user pipes stderr to another location but not stdout Rich's
    FileProxy would still steal stderr and redirect to stdout.
    
    So `guidellm run ... 2> console.log` will result in all logging ending
    up in console.log rather then only the few logs that occur before the
    FileProxy. Interestingly because we ask the OS to tell us if they are
    the same file `2>&1` keeps the FileProxy redirection in place (Which is
    what we want).
    
    Signed-off-by: Samuel Monson <smonson@redhat.com>

commit 0ee6006ff35f9a557c0cc568ff5d840a6e493585
Author: Samuel Monson <smonson@redhat.com>
Date:   Wed Sep 2 16:41:01 2026 -0400

    Use a default log filename
    
    Signed-off-by: Samuel Monson <smonson@redhat.com>

commit 400333ab9c3dbe8fc2a12bef6f23a307496a9e51
Author: Samuel Monson <smonson@redhat.com>
Date:   Thu Sep 3 12:37:12 2026 -0400

    Make worker std pipe and logging configuration optional
    
    Makes the new worker features optional in order to make testing easier.
    
    Signed-off-by: Samuel Monson <smonson@redhat.com>

commit 750d66af6c086ef18a3867b734678410b91e88fd
Author: Samuel Monson <smonson@redhat.com>
Date:   Thu Sep 3 18:51:28 2026 +0000

    Fixup tests
    
    Assisted-by: Cursor
    Signed-off-by: Samuel Monson <smonson@redhat.com>

commit d64c56608ad79f61f6cb5ce2c2991e44b4c78c7d
Author: Samuel Monson <smonson@redhat.com>
Date:   Thu Sep 3 20:25:12 2026 +0000

    Add logot to help properly test loguru
    
    Logot is a testing library which is specifically designed to handle
    `loguru`. Before the log capture kept breaking due to the global state
    of logging.
    
    Generated-by: Cursor
    Signed-off-by: Samuel Monson <smonson@redhat.com>

---------

Assisted-by: Claude Code Opus 4.6
Assisted-by: Cursor
Generated-by: Cursor
Signed-off-by: Samuel Monson <smonson@redhat.com>

### CODE REVIEW

**Review by dbutenhof — DISMISSED** [review]
Some fairly arcane stuff there, but it looks OK; and I accidentally overloaded my server and started getting neatly scrolling 503 backend exception error messages in addition to the expected DEBUG ... so looks like it works well.

**Comment by sjmonson** [comment]
Putting back to draft because there are a few more transient test issues I have uncovered.

**Review by jaredoconnell — COMMENTED** [review]
This is working well for me. I will submit my final review once it's out of draft.

**Review by dbutenhof — APPROVED** [review]
OK -- seems to work, except for that pesky integration test failure.

> AssertionError: assert 'cancelled' in {'in_progress', 'queued', 'completed', 'pending'}

I'm assuming that's the symptom Jared's attacking ...

**Review by jaredoconnell — APPROVED** [review]
Looks good to me. There are still situations where logging can still end up below the interactive console, unfortunately. But that can be addressed separately.
